0000-event-system.md

```diff
@@ -0,0 +1,140 @@
+* Start Date: 2018-01-16
+* RFC PR: (leave this empty)
+* React Navigation Issue: (leave this empty)
+
+# Summary
+
+Add an event emitter to the navigation prop that emits focus/blur events for the route.
+
+This proposal has been pioneered by @satya164: https://github.com/react-navigation/react-navigation/issues/1363 https://github.com/react-navigation/react-navigation/issues/51
+
+# Basic example
+
+In this example, we focus the first text input when the screen has completed transitioning:
+
+```
+componentDidMount() {
+  this._sub = this.props.navigation.addEventListener(
```

Comment:
Maybe `addListener` fits better because of several React Native APIs such as `Keyboard`, `Animated` etc. using `addListener`. It also signals that the behaviour is different from web's `addEventListener` which doesn't return a subscription object.

Reply:
alas `addEventListener` is used by `NetInfo` and `AppState`, there is no consistency in the world, sigh